sdk/core/core/inc/az_curl_adapter.h

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef AZ_CURL_ADAPTER
+#define AZ_CURL_ADAPTER
+
+#include <stdio.h>
+#include <curl/curl.h>
+#include <az_span.h>
+
+CURL *curl;
+
+az_result az_http_client_init_impl() {
+    curl = curl_easy_init();
+    curl_easy_setopt(curl, CURLOPT_FAILONERROR, 1l);
+    return AZ_OK;
+}
+
+az_result az_http_client_set_URL_impl(az_const_span span) {
+    curl_easy_setopt(curl, CURLOPT_URL, span.begin);
+}
+
+az_result az_http_client_download_to_file_impl(az_const_span path) {
```

Comment:
I'd move long functions into a .c file.The ones that stay in headers should be AZ_INLINE.
I'd also move global variable into a .c file. But I think it is better to redesign in a way that we accept an extra parameter in all/most of these functions, but there is no global state.